features/javascript/request.feature

```diff
@@ -0,0 +1,63 @@
+@javascript
+Feature: Request validation
+
+  Background:
+    Given you call:
+    """
+    const gavel = require('gavel');
+    """
+    And you define following HTTP request object:
+    """
+    const real = {
+      "method": "GET",
+      "uri": "/ip",
+      "headers": {
+        "user-agent": "curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8x zlib/1.2.5",
+        "host": "httpbin.org",
+        "accept": "*/*"
+      },
+      "body": ""
+    };
+    """
+    And you define following expected HTTP request object:
+    """
+    const expected = {
+      "headers": {
+        "user-agent": "curl/7.24.0 (x86_64-apple-darwin12.0) libcurl/7.24.0 OpenSSL/0.9.8x zlib/1.2.5",
+        "host": "httpbin.org",
+        "accept": "*/*"
+      },
+      "body": ""
+    };
+    """
+
+  @stable
+  Scenario: validate
+    When you call:
+    """
+    gavel.validate(expected, real);
+    """
+    Then it will return:
+    """
+    {
+      isValid: true,
+      fields: {
+        headers: {
+          isValid: true,
+          errors: [],
+          realType: 'application/vnd.apiary.http-headers+json',
```

Comment:
Should this be in the interface for Gavel? It seems an implementation detail (although perhaps I mis-understand how Gavel is used).

Reply:
It is an implementation detail indeed. I think per my previous comment, the whole `javascript` directory should really live in `gavel.js` repo. This test would assert the expected output of the `gavel.js`.